### PR TITLE
Charger modifications

### PR DESCRIPTION
**Pull request by danmois**
## Pull Request Description

Add the code for enabling operation in 120V power mains.
Make DEBUG messages display default.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: projects/ad-acevsecrdset-sl/Makefile [hunk 1]
```diff
@@ -3,3 +3,5 @@ include ../../tools/scripts/generic_variables.mk
 include src.mk
 
 include ../../tools/scripts/generic.mk
+
+CFLAGS += -DNO_OS_LOG_LEVEL=NO_OS_LOG_DEBUG
```
Comment (amiclaus) [inline projects/ad-acevsecrdset-sl/Makefile:7]: is this mandatory for all project builds? If users want explicit debug log then they should call this flag on their custom build.

File: projects/ad-acevsecrdset-sl/src/self_test/self_test.c [hunk 2]
```diff
@@ -224,7 +235,7 @@ int self_test_rcd(struct stout *stout)
 	if (ret)
 		return ret;
 
-	no_os_mdelay(40);
```
Comment (amiclaus) [inline projects/ad-acevsecrdset-sl/src/self_test/self_test.c:227]: is this a related change to what it is described in the commit body/title? (also adding a comment on the meaning of this delay)

File: projects/ad-acevsecrdset-sl/src/self_test/self_test.c [hunk 3]
```diff
@@ -421,11 +432,11 @@ static int self_test_pilot(struct stout *stout)
 	// test only the high portion of the CP
 	if (((PWM_HIGH_TEST_VAL) > (pilot_convert_to_mv(get_pwm_high_val()))) &&
 	    ((PWM_HIGH_TEST_VAL_EV_CONNECTED) < pilot_convert_to_mv(get_pwm_high_val()))) {
-		// If the CP value coresponds to state A or state B value than no error
+		// If the CP value coresponds to state A or state B value then no error
+		pr_debug("TEST PILOT: Passed\n");
```
Comment (amiclaus) [inline projects/ad-acevsecrdset-sl/src/self_test/self_test.c:436]: is this a related change to what it is described in the commit body/title?

File: projects/ad-acevsecrdset-sl/src/state_machine/state_machine.c [hunk 4]
```diff
@@ -234,6 +245,7 @@ int state_machine()
 			pilot_pwm_timer_set_duty_cycle(stout, PWM_OFF);
 			goto error;
 		}
+		stout->err_status = ret;
```
Comment (amiclaus) [inline projects/ad-acevsecrdset-sl/src/state_machine/state_machine.c:248]: is this a related change to what it is described in the commit body/title?

File: projects/ad-acevsecrdset-sl/src/state_machine/state_machine.h [hunk 5]
```diff
@@ -96,7 +96,9 @@ enum interface_err_status_e {
 	/* Overcurrent */
 	INTF_OVERCURRENT_ERR,
 	/* Overtemperature fault*/
-	INTF_TEMPERATURE_ERR
+	INTF_TEMPERATURE_ERR,
+	/*PE fault*/
+	INTF_PE_ERROR
```
Comment (amiclaus) [inline projects/ad-acevsecrdset-sl/src/state_machine/state_machine.h:101]: ditto.

File: projects/ad-acevsecrdset-sl/Makefile (hunk 1 above)
Comment (danmois) [reply projects/ad-acevsecrdset-sl/Makefile:7]: This is what we actually wanted to achieve, to make DEBUG as default for all project builds.

File: projects/ad-acevsecrdset-sl/src/state_machine/state_machine.c (hunk 4 above)
Comment (danmois) [reply projects/ad-acevsecrdset-sl/src/state_machine/state_machine.c:248]: Yes, this change is related to the change in the commit message.

File: projects/ad-acevsecrdset-sl/src/self_test/self_test.c (hunk 2 above)
Comment (danmois) [reply projects/ad-acevsecrdset-sl/src/self_test/self_test.c:227]: We created a separate commit.

File: projects/ad-acevsecrdset-sl/src/self_test/self_test.c (hunk 3 above)
Comment (danmois) [reply projects/ad-acevsecrdset-sl/src/self_test/self_test.c:436]: We created a separate commit.

File: projects/ad-acevsecrdset-sl/src/state_machine/state_machine.h (hunk 5 above)
Comment (danmois) [reply projects/ad-acevsecrdset-sl/src/state_machine/state_machine.h:101]: We created a separate commit.

**Comment by danmois** [comment]
V2

- split commit according to implemented modifications

**Review by amiclaus — APPROVED** [review]

**Review by buha — APPROVED** [review]

**Pull request merged**